addons/binding/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/internal/protocol/xml/ZoneControlXML.java

```diff
@@ -0,0 +1,209 @@
+        if (volume < 0) {
+            volume = 0;
+        }
+        if (volume > 100) {
+            volume = 100;
+        }
+        // Compute value in db
+        setVolumeDB(volume * YamahaReceiverBindingConstants.VOLUME_RANGE / 100.0f
+                + YamahaReceiverBindingConstants.VOLUME_MIN);
+    }
+
+    /**
+     * Increase or decrease the volume by the given percentage.
+     *
+     * @param percent
+     * @throws IOException
+     */
+    @Override
+    public void setVolumeRelative(ZoneControlState state, float percent)
+            throws IOException, ReceivedMessageParseException {
+        setVolume(state.volume + percent);
+    }
+
+    @Override
+    public void setMute(boolean mute) throws IOException, ReceivedMessageParseException {
+        if (mute) {
+            comReference.get().send(XMLUtils.wrZone(zone, "<Volume><Mute>On</Mute></Volume>"));
+        } else {
+            comReference.get().send(XMLUtils.wrZone(zone, "<Volume><Mute>Off</Mute></Volume>"));
+        }
+        update();
+    }
+
+    @Override
+    public void setInput(String name) throws IOException, ReceivedMessageParseException {
+        // ToDo: See the fixme in the class description
+        comReference.get().send(XMLUtils.wrZone(zone, "<Input><Input_Sel>" + name + "</Input_Sel></Input>"));
+        update();
+    }
+
+    @Override
+    public void setSurroundProgram(String name) throws IOException, ReceivedMessageParseException {
+        if (name.toLowerCase().equals("straight")) {
+            comReference.get().send(XMLUtils.wrZone(zone,
+                    "<Surround><Program_Sel><Current><Straight>On</Straight></Current></Program_Sel></Surround>"));
+        } else {
+            comReference.get().send(XMLUtils.wrZone(zone, "<Surround><Program_Sel><Current><Sound_Program>" + name
+                    + "</Sound_Program></Current></Program_Sel></Surround>"));
+        }
+
+        update();
+    }
+
+    @Override
+    public void update() throws IOException, ReceivedMessageParseException {
+        if (observer == null) {
+            return;
+        }
+
+        AbstractConnection com = comReference.get();
+        String response = com.sendReceive(XMLUtils.wrZone(zone, "<Basic_Status>GetParam</Basic_Status>"));
+        Document doc = XMLUtils.xml(response);
+        if (doc.getFirstChild() == null) {
+            throw new ReceivedMessageParseException("<Basic_Status>GetParam failed: " + response);
+        }
+        Node basicStatus = XMLUtils.getNode(doc.getFirstChild(), zone + "/Basic_Status");
+
+        String value;
+
+        ZoneControlState state = new ZoneControlState();
+
+        value = XMLUtils.getNodeContentOrDefault(basicStatus, "Power_Control/Power", "");
+        state.power = "On".equalsIgnoreCase(value);
+
+        value = XMLUtils.getNodeContentOrDefault(basicStatus, "Input/Input_Sel", "");
+        // ToDo: See the fixme in the class description
```

Comment:
@zarusz I don't have a real yamaha here to test. Would be awesome if you could adapt the code to make it work for your receiver. I'm pretty sure, the json protocol will do it different again and hopefully more logically. So shouldn't be a problem for all 2013+ models anyway.